src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -484,6 +484,19 @@ async Task<DurableOrchestrationStatus> IDurableOrchestrationClient.GetStatusAsyn
             return await GetDurableOrchestrationStatusAsync(state, this.client, showHistory, showHistoryOutput);
         }
 
+        /// <inheritdoc />
+        async Task<List<DurableOrchestrationStatus>> IDurableOrchestrationClient.GetStatusAsync(IEnumerable<string> instanceIds, bool showHistory, bool showHistoryOutput, bool showInput)
+        {
+            var results = new List<DurableOrchestrationStatus>();
+            var tasks = instanceIds.Select(async instanceId =>
+            {
+                var status = await ((IDurableOrchestrationClient)this).GetStatusAsync(instanceId);
+                results.Add(status);
+            });
+            await Task.WhenAll(tasks);
+            return results;
+            }
```

Comment:
I think this code can be simplified with some C# magic.
I suggest rewriting it like this:

```suggestion
        async Task<IList<DurableOrchestrationStatus>> IDurableOrchestrationClient.GetStatusAsync(IEnumerable<string> instanceIds, bool showHistory, bool showHistoryOutput, bool showInput)
        {
            var tasks = instanceIds.Select(instanceId => ((IDurableOrchestrationClient)this).GetStatusAsync(instanceId));
            var results = await Task.WhenAll(tasks);
            return results;
        }
```

Note that, for this to work, you'll need to change the definition of `GetStatusAsync` in `IDurableOrchestrationClient.cs` to return an `IList` instead of a `List`.

Let me know what you think!